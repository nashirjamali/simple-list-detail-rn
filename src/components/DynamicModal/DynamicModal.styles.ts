import {StyleSheet} from 'react-native';

import {COLORS} from '../../constants';
import type {DynamicModalStyles} from '../../@types';

const styles: DynamicModalStyles = StyleSheet.create({
  overlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalContainer: {
    backgroundColor: COLORS.WHITE,
    borderRadius: 10,
    padding: 20,
    width: '80%',
  },
});

export default styles;
