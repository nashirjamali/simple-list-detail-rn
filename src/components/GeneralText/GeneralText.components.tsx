import React, {FC} from 'react';
import {Text, type TextStyle} from 'react-native';

import type {GeneralTextProps, GeneralTextVariantsTypes} from '../../@types';

import styles from './GeneralText.styles';

/**
 * Retrieves the text style based on the provided variant.
 *
 * @param {GeneralTextVariantsTypes} variant - The variant type for the text style.
 * @returns {TextStyle} The corresponding text style.
 */
const getTextStyle = (variant: GeneralTextVariantsTypes): TextStyle => {
  return styles[variant];
};

/**
 * A general text component that applies styles based on the variant.
 *
 * @param {GeneralTextProps} props - The properties for the GeneralText component.
 * @param {GeneralTextVariantsTypes} props.variant - The variant type for styling.
 * @param {React.ReactNode} props.children - The content to be displayed within the text.
 * @returns {JSX.Element} The rendered text component.
 */
const GeneralText: FC<GeneralTextProps> = ({
  variant,
  children,
  ...props
}: GeneralTextProps): JSX.Element => (
  <Text {...getTextStyle(variant)} {...props}>
    {children}
  </Text>
);

export default GeneralText;
