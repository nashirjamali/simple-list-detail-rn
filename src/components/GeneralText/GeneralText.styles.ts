import {StyleSheet} from 'react-native';

import {COLORS} from '../../constants';
import type {GeneralTextStyles} from '../../@types';

const {BLACK, BRAND, GRAY} = COLORS;

const styles: GeneralTextStyles = StyleSheet.create({
  button: {
    color: BRAND,
    fontSize: 18,
  },
  header: {
    color: BLACK,
    fontSize: 20,
    fontWeight: 'bold',
  },
  input: {
    color: GRAY,
    fontSize: 16,
  },
  label: {
    fontSize: 16,
  },
  normal: {
    color: BLACK,
    fontSize: 16,
  },
});

export default styles;
