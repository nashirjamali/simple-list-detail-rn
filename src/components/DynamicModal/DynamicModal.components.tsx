import React, {FC} from 'react';
import {Modal, View, Pressable} from 'react-native';

import type {DynamicModalProps} from '../../@types';
import {GENERAL_TEXT_VARIANTS} from '../../constants';

import GeneralText from '../GeneralText/GeneralText.components';

import styles from './DynamicModal.styles';

/**
 * A customizable modal component that displays content over the app.
 *
 * @param {DynamicModalProps} props - The properties for the DynamicModal component.
 * @param {boolean} props.visible - Determines if the modal is visible.
 * @param {string} props.title - The title to display at the top of the modal.
 * @param {() => void} props.onClose - Function to call when the modal is closed.
 * @param {React.ReactNode} props.children - The content to display inside the modal.
 * @returns {JSX.Element} The rendered modal component.
 */
const DynamicModal: FC<DynamicModalProps> = ({
  visible,
  title,
  onClose,
  children,
}) => {
  return (
    <Modal
      transparent={true}
      visible={visible}
      animationType="slide"
      onRequestClose={onClose}>
      <Pressable style={styles.overlay} onPress={onClose}>
        <View style={styles.modalContainer}>
          <GeneralText variant={GENERAL_TEXT_VARIANTS.HEADER}>
            {title}
          </GeneralText>
          {children}
        </View>
      </Pressable>
    </Modal>
  );
};

export default DynamicModal;
