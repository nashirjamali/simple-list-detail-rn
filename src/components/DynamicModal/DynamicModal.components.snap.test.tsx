import React from 'react';
import {render} from '@testing-library/react-native';

import GeneralText from '../GeneralText/GeneralText.components';

import DynamicModal from './DynamicModal.components';

describe('DynamicModal Component', () => {
  it('renders correctly when visible', () => {
    const {toJSON} = render(
      <DynamicModal visible={true} title="Test Modal" onClose={jest.fn()}>
        <GeneralText variant="normal">
          This is a test modal content.
        </GeneralText>
      </DynamicModal>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when not visible', () => {
    const {toJSON} = render(
      <DynamicModal visible={false} title="Test Modal" onClose={jest.fn()}>
        <GeneralText variant="normal">
          This is a test modal content.
        </GeneralText>
      </DynamicModal>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
