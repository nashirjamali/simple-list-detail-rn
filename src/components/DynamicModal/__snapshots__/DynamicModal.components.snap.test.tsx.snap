// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicModal Component renders correctly when not visible 1`] = `null`;

exports[`DynamicModal Component renders correctly when visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "padding": 20,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#121212",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Test Modal
      </Text>
      <Text
        style={
          {
            "color": "#121212",
            "fontSize": 14,
          }
        }
      >
        This is a test modal content.
      </Text>
    </View>
  </View>
</Modal>
`;
